File: LocalisationMateriel/src/java/metier/MetierFactory.java
package metier;


public class MetierFactory {
    
    private static BatimentService batimentSrv = null;
    private static LecteurRFIDService lecteurRFIDSrv = null;
    private static LocalisationService localisationSrv = null;
    private static MaterielService materielSrv = null;
    private static MetrologueService metrologueSrv = null;
    private static PortiqueService portiqueSrv = null;
    private static SalleService salleSrv = null;
    private static TypeService typeSrv = null;

    public static BatimentService getBatimentService() {
        if(batimentSrv == null){
            batimentSrv = new BatimentServiceImpl();
        }
        return batimentSrv;
    }
    
    public static LecteurRFIDService getLecteurRFIDService(){
        if(lecteurRFIDSrv == null){
            lecteurRFIDSrv = new LecteurRFIDServiceImpl();
        }
        return lecteurRFIDSrv;
    }
    
    public static LocalisationService getLocalisationService(){
        if(localisationSrv == null){
            localisationSrv = new LocalisationServiceImpl();
        }
        return localisationSrv;
    }
    
    public static MaterielService getMaterielService() {
        if (materielSrv == null) {
            materielSrv = new MaterielServiceImpl();
        }
        return materielSrv;
    }
    
    public static MetrologueService getMetrologueService(){
        if(metrologueSrv == null){
            metrologueSrv = new MetrologueServiceImpl();
        }
        return metrologueSrv;
    }
    
    public static PortiqueService getPortiqueService(){
        if(portiqueSrv == null){
            portiqueSrv = new PortiqueServiceImpl();
        }
        return portiqueSrv;
    }
    
    public static SalleService getSalleService(){
        if(salleSrv == null){
            salleSrv = new SalleServiceImpl();
        }
        return salleSrv;
    }
    
    public static TypeService getTypeService() {
        if (typeSrv == null) {
            typeSrv = new TypeServiceImpl();
        }
        return typeSrv;
    }
}

File: AppMateriel/metier/metierfactory.cpp
#include "metierfactory.h"

Metier::BatimentService* MetierFactory::batimentSrv = 0;
Metier::LocalisationService* MetierFactory::localisationSrv = 0;
Metier::MaterielService* MetierFactory::materielSrv = 0;
Metier::MetrologueService* MetierFactory::metrologueSrv = 0;
Metier::PortiqueService* MetierFactory::portiqueSrv = 0;
Metier::SalleService* MetierFactory::salleSrv = 0;
Metier::TypeService* MetierFactory::typeSrv = 0;
Metier::LecteurRFIDService* MetierFactory::lecteurRFIDSrv = 0;

Metier::BatimentService* MetierFactory::getBatimentServiceImpl(){
    if(batimentSrv == 0){
        batimentSrv = new Metier::BatimentServiceImpl();
    }
    return batimentSrv;
}

Metier::LocalisationService* MetierFactory::getLocalisationServiceImpl(){
    if(localisationSrv == 0){
        localisationSrv = new Metier::LocalisationServiceImpl();
    }
    return localisationSrv;
}

Metier::MaterielService* MetierFactory::getMaterielServiceImpl(){
    if(materielSrv == 0){
        materielSrv = new Metier::MaterielServiceImpl();
    }
    return materielSrv;
}

Metier::MetrologueService* MetierFactory::getMetrologueServiceImpl(){
    if(metrologueSrv == 0){
        metrologueSrv = new Metier::MetrologueServiceImpl();
    }
    return metrologueSrv;
}

Metier::PortiqueService* MetierFactory::getPortiqueServiceImpl(){
    if(portiqueSrv == 0){
        portiqueSrv = new Metier::PortiqueServiceImpl();
    }
    return portiqueSrv;
}

Metier::SalleService* MetierFactory::getSalleServiceImpl(){
    if(salleSrv == 0){
        salleSrv = new Metier::SalleServiceImpl();
    }
    return salleSrv;
}

Metier::TypeService* MetierFactory::getTypeServiceImpl(){
    if(typeSrv == 0){
        typeSrv = new Metier::TypeServiceImpl();
    }
    return typeSrv;
}

Metier::LecteurRFIDService *MetierFactory::getLecteurRFIDServiceImpl()
{
    if(lecteurRFIDSrv == 0){
        lecteurRFIDSrv = new Metier::LecteurRFIDServiceImpl();
    }
    return lecteurRFIDSrv;
}
